test/integration/tasks.test.js

```diff
@@ -166,6 +180,83 @@ describe("Tasks", function () {
           return done();
         });
     });
+
+    it(`Should get all tasks except ${TASK_STATUS.IN_PROGRESS} tasks`, function (done) {
+      chai
+        .request(app)
+        .get(`/tasks/?q=filter%3A${TASK_STATUS.IN_PROGRESS}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Tasks returned successfully!");
+          expect(res.body.tasks).to.be.a("array");
+          expect(res.body.tasks.filter((task) => task.status === TASK_STATUS.IN_PROGRESS).length).to.be.eql(0);
+          return done();
+        });
+    });
+
+    it(`Should only get the ${TASK_STATUS.IN_PROGRESS} tasks`, function (done) {
+      chai
+        .request(app)
+        .get(`/tasks/?q=type%3A${TASK_STATUS.IN_PROGRESS}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Tasks returned successfully!");
+          expect(res.body.tasks).to.be.a("array");
+          expect(res.body.tasks.some((task) => task.status !== TASK_STATUS.IN_PROGRESS)).to.be.eql(false);
+          return done();
+        });
+    });
+
+    it(`Should get all tasks except ${TASK_STATUS.IN_PROGRESS}, ${TASK_STATUS.BLOCKED} tasks`, function (done) {
+      chai
+        .request(app)
+        .get(`/tasks/?q=filter%3A${TASK_STATUS.IN_PROGRESS}%2C${TASK_STATUS.BLOCKED}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Tasks returned successfully!");
+          expect(res.body.tasks).to.be.a("array");
+          expect(
+            res.body.tasks.filter((task) => [TASK_STATUS.IN_PROGRESS, TASK_STATUS.BLOCKED].includes(task.status)).length
+          ).to.be.eql(0);
+          expect(res.body.tasks.length).to.be.eq(2);
+          return done();
+        });
+    });
+
+    it(`Should only get the ${TASK_STATUS.IN_PROGRESS} which startsAfter cursor:${taskId1} tasks`, async function () {
+      const response = await chai.request(app).get(`/tasks/?q=type%3A${TASK_STATUS.IN_PROGRESS}`);
+
+      expect(response).to.have.status(200);
+      expect(response.body).to.be.a("object");
+      expect(response.body.message).to.equal("Tasks returned successfully!");
+      expect(response.body.tasks).to.be.a("array");
+      expect(response.body.tasks.length).to.be.eq(2);
+
+      const firstTaskId = response.body.tasks[0].id;
+      const nextTaskId = response.body.tasks[1].id;
+      const nextResponse = await chai
+        .request(app)
+        .get(`/tasks/?q=type%3A${TASK_STATUS.IN_PROGRESS}+cursor%3A${firstTaskId}`);
```

Comment:
can we add one more task to which starts before `firstTaskId ` to be double sure of this test

Reply:
What I have understood is you want to add one more task of type `progress` and run the query **startsBefore**, which I don't think will be possible.
Else what I have done is add one more task and requested for tasks after index item `1`.

Correct me, if I understood it wrong.

Reply:
I console logged the response body 
```js
{
  message: 'Tasks returned successfully!',
  tasks: [
    {
      id: '03oBjpQyvgglJ6gIVuqQ',
      percentCompleted: 10,
      endsOn: 1234,
      isNoteworthy: true,
      lossRate: [Object],
      assignee: 'sagar',
      title: 'Test task',
      type: 'feature',
      completionAward: [Object],
      startedOn: 4567,
      participants: [],
      status: 'IN_PROGRESS'
    },
    {
      id: '7BjJ81wbfMfMrdt1ZMZV',
      percentCompleted: 10,
      endsOn: 1234,
      isNoteworthy: true,
      lossRate: [Object],
      assignee: 'sagar',
      title: 'Some Task 4',
      type: 'feature',
      completionAward: [Object],
      startedOn: 4567,
      participants: [],
      status: 'IN_PROGRESS'
    },
    {
      id: 'L2ElETtolXZVEbYfyhkH',
      percentCompleted: 10,
      endsOn: 1234,
      isNoteworthy: true,
      lossRate: [Object],
      assignee: 'sagar',
      title: 'Some Task 3',
      type: 'feature',
      completionAward: [Object],
      startedOn: 4567,
      participants: [],
      status: 'IN_PROGRESS'
    }
  ],
  cursor: 'L2ElETtolXZVEbYfyhkH'
}
```
All three tasks have the same `starts On` 🤔